hamp_radar/iquick.py

```diff
@@ -186,28 +340,51 @@ def _decode(rawdata):
 
 
 def decode_iq(rawdata):
-    # TODO HACK: the 256 (for nfft) just appears, it probably should be read from somewhere else in the data
+    # TODO(ALL) HACK: the 256 (for nfft) just appears, it probably should be read from somewhere else in the data
     return {
         "FFTD": (
             ("frame", "range", "cocx", "fft", "iq"),
             rawdata.view("<i2").reshape(rawdata.shape[0], -1, 2, 256, 2),
         )
     }
 
-
+#TODO(ALL): move decoders to seperate file
 decoders = {
+    """
+    Decoders for IQ data as in Meteorological Ka-Band Cloud Radar MIRA35 Manual,
+    section 2.3.3.2 'Embedded chain type 2; Data chain'. Note these decoders are
+    specific to the Ka radar currently in operation on HALO.
+    (last checked: 13th Septermber 2024).
+    """
     b"SRVI": decode_srvi,
     b"SNRD": decode_moment("SNRD"),
     b"VELD": decode_moment("VELD"),
     b"HNED": decode_moment("HNED"),
     b"RMSD": decode_moment("RMSD"),
-    b"FFTD": decode_iq,  # TODO HACK: FFTD may or may not be IQ data. This is configured in PPAR
+    b"FFTD": decode_iq,  # TODO(ALL) HACK: FFTD may or may not be IQ data. This is configured in PPAR
 }
 
 
-def read_iq(filename, postprocess=True):
-    data = np.memmap(filename)
-    raw_arrays = list(extract_raw_arrays(data, get_geometry(data)))
+def read_pds(filename, postprocess=True):
+    """
+    Converts data from a file called 'filename', into an xarray Dataset. Currently
+    only functioning with geometry of pds files and decoders for IQ data of
+    Ka radar currently operational on HALO (last checked: 13th Septermber 2024).
+
+    Optimisation uses NumPy library memory-mapped array to avoid reading
+    entirety of large binary file into main memory from disc when only (small)
+    segment is desired.
```

Comment:
I don't have a super strong opinion here, but this also is an implementation detail, not super relevant to external users.
```suggestion
```